MSVC build broken

pycurl.h now has this:
```
#  warning \
  "libcurl was compiled with SSL support, ...
```

This is no good for MSVC. Should be:
```
# ifdef _MSC_VER
#   pragma message("libcurl was compiled with SSL support ...")
#  else
```

Can I extend a closed PR instead of creating a new? Good, here is another MSVC breakage from today's GitHub:
```
python setup.py build --curl-dir=..
Using curl directory: ..
running build
running build_py
....

f:\ProgramFiler-x86\VS2013\VC\bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -DPYCURL_VERSION="7.19.5.1" -I..\include -If:\ProgramFiler\Pyt
hon27\include -If:\ProgramFiler\Python27\PC /Tcsrc\oscompat.c /Fobuild\temp.win32-2.7\Release\src\oscompat.obj -DCURL_STATICLIB -D_WIN32_WIN
NT=0x0501 -O2 -GF -WX
oscompat.c
src\oscompat.c(10) : error C2220: warning treated as error - no 'object' file generated
src\oscompat.c(10) : warning C4996: 'WSADuplicateSocketA': Use WSADuplicateSocketW() instead or define _WINSOCK_DEPRECATED_NO_WARNINGS to disable deprecated API warnings
```
Pretty clear? I'm on Win-8.1 using MS-SDK 8.1.
A `C:\>set CL=-D_WINSOCK_DEPRECATED_NO_WARNINGS` fixes it.

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,6 +1,9 @@
 master
 ------
 
+        * Fixed build on Windows when using MS SDK 8.1+ or MSVC 14/2015
+          (patch by Gisle Vanem).
+
         * Added automatic SSL library detection on CentOS 6 by loading
           libcurl shared library in setup.py. This automatic detection is
           meant to permit installing pycurl seamlessly via `pip install pycurl`

--- a/src/pycurl.h
+++ b/src/pycurl.h
@@ -16,14 +16,16 @@
 #include <arpa/inet.h>
 #endif
 
+#if defined(WIN32)
 /*
  * Since setup.py uses a '-WX' in the CFLAGS (treat warnings as errors),
  * the below will turn off some warnings when using MS-SDK 8.1+.
  * This MUST be defined before including <winsock2.h> via the libcurl
  * headers.
  */
-#ifndef _WINSOCK_DEPRECATED_NO_WARNINGS
-#define _WINSOCK_DEPRECATED_NO_WARNINGS
+# if !defined(_WINSOCK_DEPRECATED_NO_WARNINGS)
+#  define _WINSOCK_DEPRECATED_NO_WARNINGS
+# endif
 #endif
 
 #include <curl/curl.h>

--- a/src/pycurl.h
+++ b/src/pycurl.h
@@ -16,6 +16,16 @@
 #include <arpa/inet.h>
 #endif
 
+/*
+ * Since setup.py uses a '-WX' in the CFLAGS (treat warnings as errors),
+ * the below will turn off some warnings when using MS-SDK 8.1+.
+ * This MUST be defined before including <winsock2.h> via the libcurl
+ * headers.
+ */
+#ifndef _WINSOCK_DEPRECATED_NO_WARNINGS
+#define _WINSOCK_DEPRECATED_NO_WARNINGS
+#endif
+
 #include <curl/curl.h>
 #include <curl/easy.h>
 #include <curl/multi.h>

